src/globals.ts

```diff
@@ -95,10 +101,15 @@ export const ExecutionTags = (() => {
 
   const addTag = (key, value, shouldLogErrors = true) => {
     try {
-      logger.debug(`Adding tag: ${key} - ${value}`);
-      if (!validateTag(key, value, shouldLogErrors)) return false;
-      // @ts-ignore
-      global.tags.push({ key: normalizeTag(key), value: normalizeTag(value) });
+      if (isLambdaTraced()) {
+        logger.debug(`Adding tag: ${key} - ${value}`);
+        if (!validateTag(key, value, shouldLogErrors)) return false;
+        // @ts-ignore
+        global.tags.push({ key: normalizeTag(key), value: normalizeTag(value) });
+      } else {
+        shouldLogErrors && logger.warnClient(`${ADD_TAG_ERROR_MSG_PREFIX}: lambda is not traced`);
+        return false;
+      }
     } catch (err) {
       shouldLogErrors && logger.warnClient(ADD_TAG_ERROR_MSG_PREFIX);
```

Comment:
do you know why the error isn't included in the client warning?

Reply:
we show the client only specific errors, a general catch , will contain information that the client don't really need to see